src/renderer/components/_molecules/menu-bar/menus/file.tsx

```diff
@@ -10,35 +10,38 @@ import { useEffect } from 'react'
 import { useNavigate } from 'react-router-dom'
 
 import { MenuClasses } from '../constants'
+import { SaveChangesModal } from '../modals/save-changes-modal'
 
 export const FileMenu = () => {
   const navigate = useNavigate()
   const {
     project,
+    workspace: { editingState },
     editorActions: { clearEditor },
     workspaceActions: { setEditingState, setRecents },
     projectActions: { setProject },
     tabsActions: { clearTabs },
     flowActions: { addFlow },
     editor,
+    modals,
     modalActions: { openModal },
   } = useOpenPLCStore()
   const { handleRemoveTab, selectedTab, setSelectedTab } = useHandleRemoveTab()
   const { TRIGGER, CONTENT, ITEM, ACCELERATOR, SEPARATOR } = MenuClasses
 
   const handleCreateProject = () => {
-    try {
+    if (editingState === 'unsaved') {
+      openModal('save-changes-project', 'create-project')
+    } else {
       openModal('create-project', null)
-    } catch (_error) {
-      toast({
-        title: 'Error',
-        description: 'Failed to open project creation modal',
-        variant: 'fail',
-      })
     }
   }
 
   const handleOpenProject = async () => {
+    if (editingState === 'unsaved') {
+      openModal('save-changes-project', 'open-project')
+      return
+    }
```

Comment:
Keep handleOpenProject and handleCreateProject separate, even if they share similar code snippets. This is because the purposes of each function are clearly distinct:

handleCreateProject: Responsible for starting a new project. It focuses on preparing the environment for an empty project or handling any necessary initial setup.

handleOpenProject: Handles opening an existing project. This involves loading data from a file, setting up the editor state, and performing other actions specific to a previously saved project.